src/main/groovy/org/shipkit/internal/notes/contributors/ProjectContributorFetcherFunction.java

```diff
@@ -0,0 +1,37 @@
+package org.shipkit.internal.notes.contributors;
+
+import org.json.simple.DeserializationException;
+import org.json.simple.JsonObject;
+import org.shipkit.internal.notes.model.ProjectContributor;
+import org.shipkit.internal.notes.util.Function;
+import org.shipkit.internal.notes.util.GitHubObjectFetcher;
+
+import java.io.IOException;
+
+/**
+ * Extracts the url form a given {@link JsonObject} and fetches the {@link ProjectContributor} info using the extracted
+ * url.
+ */
+class ProjectContributorFetcherFunction implements Function<JsonObject, ProjectContributor> {
+
+    private final GitHubObjectFetcher objectFetcher;
+
+    public ProjectContributorFetcherFunction(GitHubObjectFetcher objectFetcher) {
+        this.objectFetcher = objectFetcher;
+    }
+
+    @Override
+    public ProjectContributor apply(JsonObject contributor) {
+        String url = (String) contributor.get("url");
+        JsonObject user;
+        try {
+            user = objectFetcher.getPage(url);
+        } catch (IOException e) {
```

Comment:
Single catch clause with 'Exception' gives us simpler code and less duplication.